service/dynamodb/internal/customizations/accept_encoding_gzip.go

```diff
@@ -79,7 +79,7 @@ func (*AcceptEncodingGzipMiddleware) HandleFinalize(
 ) {
 	req, ok := input.Request.(*smithyhttp.Request)
 	if !ok {
-		return output, metadata, &smithy.SerializationError{
+		return output, metadata, &smithy.DeserializationError{
```

Comment:
This and the one above, should be `SerializationError` Since it is on Request, not response